src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBoxChosenSuggestion.xaml.cs

```diff
@@ -0,0 +1,149 @@
+
+namespace UITests.Shared.Windows_UI_Xaml_Controls.AutoSuggestBoxTests
+{
+	public class Author
+	{
+		public static Author[] All = new Author[]
+											{
+														new Author { Name = "A0" },
+														new Author { Name = "A1" },
+														new Author { Name = "A2" },
+														new Author { Name = "A3" },
+														new Author { Name = "B0" },
+														new Author { Name = "B1" },
+														new Author { Name = "B2" },
+														new Author { Name = "B3" },
+														new Author { Name = "a0" },
+														new Author { Name = "a1" },
+														new Author { Name = "a2" },
+														new Author { Name = "a3" },
+											};
+
+		public string Name
+		{ get; set; } = string.Empty;
+
+		public override string ToString()
+		{
+			return Name;
+		}
+	}
+
+	public class AutoSuggestConverter : IValueConverter
+	{
+		public object Convert(object value, Type targetType, object parameter, string language)
+		{
+			if (value == null)
+				return null;
+
+			return value.ToString();
+		}
+
+		public object ConvertBack(object value, Type targetType, object parameter, string language)
+		{
+			return ((parameter as AutoSuggestBox)?.ItemsSource as IList)?.Cast<object>().FirstOrDefault(i => i.ToString() == (value as string));
+		}
+	}
+
+	public class Book
+	{
+		private Author author;
+
+		public Author Author
+		{
+			get => author;
+			set => author = value;
+		}
+	}
+
+    public static class Extensions
+    {
+        public static TElement Bind<TElement>(this TElement element, object source, string path = null, IValueConverter converter = null, object converterParameter = null, BindingMode mode = BindingMode.TwoWay) where TElement : FrameworkElement
+        {
+            element.SetBinding(AutoSuggestBox.TextProperty, new Binding
+												{
+													Converter			= converter,
+													ConverterParameter	= converterParameter,
+													Mode				= mode,
+													Path				= string.IsNullOrEmpty(path) ? null : new PropertyPath(path),
+													Source				= source ?? element,
+												});
+            return element;
+        }
+
+        public static AutoSuggestBox OnQuerySubmitted(this AutoSuggestBox element, TypedEventHandler<AutoSuggestBox, AutoSuggestBoxQuerySubmittedEventArgs> handler)
+        {
+			element.QuerySubmitted += handler;
+			return element;
+		}
+
+        public static AutoSuggestBox OnTextChanged(this AutoSuggestBox element, TypedEventHandler<AutoSuggestBox, AutoSuggestBoxTextChangedEventArgs> handler)
+        {
+			element.TextChanged += handler;
+			return element;
+		}
+
+        public static TElement Row<TElement>(this TElement element, int row, int span = 1) where TElement : FrameworkElement
+        {
+            if (row != 0)
+				element.SetValue(Grid.RowProperty, row);
+
+            if (span != 1)
+				element.SetValue(Grid.RowSpanProperty, span);
+
+            return element;
+        }
+	}
```

Comment:
You should not need this block. Use existing techniques to write this, and move to the XAML as needed.